include/core/CSmallVector.h

```diff
@@ -142,6 +142,17 @@ class CSmallVector : public boost::container::small_vector<T, N> {
         return *this;
     }
 
+    std::string toDelimited(const std::string& delimiter = ", ") const {
+        std::string result;
+        for (size_type i = 0; i < this->size(); ++i) {
+            result += std::to_string((*this)[i]);
+            if (i < this->size() - 1) {
+                result += delimiter;
+            }
```

Comment:
As I recall appending has quite some overhead for `std::string` due to reallocations. How about something like
```
if (i < this->size() - 1) {
    result += delimiter;
}
if (i == 0) {
    result.reserve(result.size() * this->size());
}
```